### PR TITLE
Only build min and max Python version wheels on PRs

### PR DESCRIPTION
**Pull request by JCGoran**
To speed-up the CI on PRs, we only build wheels for the min (currently 3.8) and max (currently 3.12) versions of Python.

### CODE REVIEW

**Comment by 1uc** [comment]
It wasn't the last piece of CI to go green anymore.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -216,8 +216,23 @@ stages:
...
+          export TAG="-nightly"
+        else
+          export TAG=""
+        fi
+        python3 -m pip install --upgrade pip
+        python3 -m pip install cibuildwheel==2.16.5 tomli tomli-w
+        # change the name accordingly
+        python3 packaging/change_name.py pyproject.toml "NMODL${TAG}"
+        export SETUPTOOLS_SCM_PRETEND_VERSION="$(git describe --tags | cut -d '-' -f 1,2 | tr - .)"
+        CIBW_BUILD="cp38* cp312*" CIBW_ENVIRONMENT_PASS_LINUX='SETUPTOOLS_SCM_PRETEND_VERSION' python3 -m cibuildwheel --output-dir wheelhouse
```
Comment (1uc) [inline azure-pipelines.yml:232]: Can one reduce the duplication by setting `CIBW_BUILD` to one value for nightly and another value for regular PR CI runs?

**Review by 1uc — APPROVED** [review]

File: azure-pipelines.yml (hunk 1 above)
Comment (JCGoran) [reply azure-pipelines.yml]: I think I removed most of the duplication now, with two things to keep in mind (i.e. an explanation why it works):
- any azure variable not set, but used later (via `$(VARIABLE)`) will expand to an empty string
- if `CIBW_BUILD` is set to the empty string, `cibuildwheel` will ignore it and build all configurations (which is what we want for non-PRs)

**Review by 1uc — APPROVED** [review]
Is there a particular advantage of using `###vso[task....]` syntax over the standard bash command `export`?

**Comment by JCGoran** [comment]
I just read a bit on https://learn.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch#code-try-0 and `vso[...]` was shown as the way to set variables. `export` is nowhere to be found there, so I don't know if it works when you have multiple steps in a single job.

**Pull request merged**